### PR TITLE
Use log id field to identify logs

### PR DESCRIPTION
**Pull request by TurpIF**
JIRA: EE-1359

### CODE REVIEW

File: publisher-sdk/src/main/java/com/criteo/publisher/ErrorLogMessage.kt [hunk 1]
```diff
@@ -28,20 +28,23 @@ internal object ErrorLogMessage {
   inline fun onUncaughtErrorAtPublicApi(throwable: Throwable) = LogMessage(
       Log.ERROR,
       "Internal error in ${CallerInferrer.inferCallerName()}",
-      throwable
+      throwable,
+      "onUncaughtErrorAtPublicApi"
```
Comment (Amokrane) [inline publisher-sdk/src/main/java/com/criteo/publisher/ErrorLogMessage.kt:32]: What do you think about using: 
```
object{}.javaClass.enclosingMethod.name
```
(https://stackoverflow.com/questions/48181751/get-name-of-current-function-in-kotlin)
for all the logIds? That way, if the method is renamed, the log will follow as well.

File: publisher-sdk/src/main/java/com/criteo/publisher/logging/Logger.java [hunk 2]
```diff
@@ -89,15 +89,15 @@ public Logger(
...
   }
 
   public void debug(String message, Throwable thrown) {
-    log(new LogMessage(Log.DEBUG, message, thrown));
+    log(new LogMessage(Log.DEBUG, message, thrown, null));
   }
 
   public void debug(String message, Object... args) {
-    log(new LogMessage(Log.DEBUG, String.format(message, args), null));
+    log(new LogMessage(Log.DEBUG, String.format(message, args), null, null));
```
[the remark this answers is not in the document]
Comment (TurpIF) [reply publisher-sdk/src/main/java/com/criteo/publisher/logging/Logger.java:100]: This is Java. There is no such things as named params.

File: publisher-sdk/src/main/java/com/criteo/publisher/ErrorLogMessage.kt (hunk 1 above)
Comment (TurpIF) [reply publisher-sdk/src/main/java/com/criteo/publisher/ErrorLogMessage.kt:32]: I thought about this, but unfortunately, this won't work well:
- code is mangled by proguard: we would have method names like `a, b, c, ...`
- this is kotlin code which is kind of a sugar over JVM bytecode: internal methods are renamed to `methodName_packageName`, other renaming can occur
- in this case, the method in `inline`: you will get the name of the caller (or not if caller is Java) instead

This is too much issues I guess.

Also, I'm not sure we really want to rename the logId when method is renamed. But that's another discussion.

File: publisher-sdk/src/main/java/com/criteo/publisher/BiddingLogMessage.kt [hunk 3]
```diff
@@ -34,7 +34,9 @@ internal object BiddingLogMessage {
   )
 
   @JvmStatic
-  fun onInvalidAdUnit(adUnit: CacheAdUnit) = LogMessage(Log.WARN, message =
-    "Found an invalid AdUnit: $adUnit"
+  fun onInvalidAdUnit(adUnit: CacheAdUnit) = LogMessage(
+      Log.WARN,
+      "Found an invalid AdUnit: $adUnit",
+      logId = "onInvalidAdUnit"
```
[the remark this answers is not in the document]
Comment (TurpIF) [reply publisher-sdk/src/main/java/com/criteo/publisher/BiddingLogMessage.kt:40]: Done

File: publisher-sdk/src/main/java/com/criteo/publisher/logging/Logger.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Amokrane) [reply publisher-sdk/src/main/java/com/criteo/publisher/logging/Logger.java:100]: Ah yeah oops :)

**Review by Amokrane — APPROVED** [review]

**Pull request merged**